core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -374,6 +404,30 @@ private[zio] final class FiberContext[E, A](
 
   final def poll: UIO[Option[Exit[E, A]]] = ZIO.effectTotal(poll0)
 
+  final def inheritLocals: UIO[Unit] =
+    for {
+      descriptor <- ZIO.descriptor
+      locals     <- copyFiberLocals(descriptor)
+      _ <- ZIO.foreach_(locals) {
+            case (fiberRef, (value, updateFiberId)) =>
+              if (updateFiberId != descriptor.id) {
```

Comment:
What's this logic about?

Reply:
It was sort of optimization as we could skip values we know they were not changed. But it's not relevant anymore since we use `FiberRefModify` now.